Realm/RLMAnalytics.mm

```diff
@@ -194,73 +296,206 @@ static auto RLMSysCtl(int *mib, u_int mibSize, size_t *bufferSize) {
+                } else {
+                    [featuresDictionary setObject:@1 forKey:@"Primitive_List"];
+                }
+                continue;
+            }
+            if (property.set) {
+                if (property.type == RLMPropertyTypeObject) {
+                    [featuresDictionary setObject:@1 forKey:@"Reference_Set"];
+                } else {
+                    [featuresDictionary setObject:@1 forKey:@"Primitive_Set"];
+                }
+                continue;
+            }
+            if (property.dictionary) {
+                if (property.type == RLMPropertyTypeObject) {
+                    [featuresDictionary setObject:@1 forKey:@"Reference_Dictionary"];
+                } else {
+                    [featuresDictionary setObject:@1 forKey:@"Primitive_Dictionary"];
+                }
+                continue;
+            }
+
+            switch (property.type) {
+               case RLMPropertyTypeAny:
+                    [featuresDictionary setObject:@1 forKey:@"Mixed"];
+                  break;
+               case RLMPropertyTypeObject:
+                    [featuresDictionary setObject:@1 forKey:@"Reference_Link"];
+                  break;
+                case RLMPropertyTypeLinkingObjects:
+                     [featuresDictionary setObject:@1 forKey:@"Backlink"];
+                   break;
+               default:
+                    break;
+            }
+        }
+    }
+    return featuresDictionary;
+}
+
+static NSDictionary *RLMConfigurationMetrics(RLMRealmConfiguration *configuration) {
+    BOOL isSyncEnable = configuration.syncConfiguration != nil;
+    BOOL isSync = configuration.syncConfiguration != nil;
+    BOOL isPBSSync = configuration.syncConfiguration.partitionValue != nil;
+    BOOL isFlexibleSync = (isSync && !isPBSSync);
+    BOOL isCompactOnLaunch = configuration.shouldCompactOnLaunch != nil;
+    BOOL migrationBlock = configuration.migrationBlock != nil;
+
+    auto resetMode = configuration.syncConfiguration.clientResetMode;
+    return @{
+        // Sync
+        @"Sync Enabled": isSyncEnable ? @"true" : @"false",
+        @"Flexible_Sync": isFlexibleSync ? @1 : @0,
+        @"Pbs_Sync": isPBSSync ? @1 : @0,
+
+        // Client Reset
+        @"Client_Reset_Recover_Or_Discard": (isSync && resetMode == RLMClientResetModeRecoverOrDiscardUnsyncedChanges) ? @1 : @0,
+        @"Client_Reset_Recover": (isSync && resetMode == RLMClientResetModeRecoverUnsyncedChanges) ? @1 : @0,
+        @"Client_Reset_Discard": (isSync && resetMode == RLMClientResetModeDiscardUnsyncedChanges) ? @1 : @0,
+        @"Client_Reset_Manual": (isSync && resetMode == RLMClientResetModeManual) ? @1 : @0,
+
+        // Configuration
+        @"Compact_On_Launch": isCompactOnLaunch ? @1 : @0,
+        @"Schema_Migration_Block": migrationBlock ? @1 : @0,
+    };
+}
+
+void RLMSendAnalytics(RLMRealmConfiguration *configuration, RLMSchema *schema) {
```

Comment:
This is now doing quite a lot of work, so it should dispatch that work to a background queue rather than blocking opening the Realm while it prepares the analytics payload.